File: 09-09-22/Find_the_Middle_Index_in_Array.py
class Solution:
    def findMiddleIndex(self, n: List[int]) -> int:
        if(len(n)==1):
            return 0
        a=[0]
        s=0
        for i in n:
            s+=i
            a.append(s)
        maxx=a[-1]
        for i in range(0,len(a)-1):
            if(a[i]==maxx-a[i+1]):
                return i
            
            
        return -1

File: 09-09-22/Find_the_Middle_Index_in_Array.cpp
class Solution {
public:
    int findMiddleIndex(vector<int>& n) {
        int si=n.size();
        vector<int> ps ;
        ps.push_back(0);
        int sum=0;
        for(int i=0;i<si;i++)
        {
            sum+=n[i];
            ps.push_back(sum);
        }
        for (int i=0;i<si;i++)
        {
            if(ps[i]==sum-ps[i+1])
            {
                return i ;
            }
        }
        return -1;
    }
};